source/deployment/sso-saml.rst

```diff
@@ -0,0 +1,110 @@
+   username.
+-  **Automatic account provisioning.** New Mattermost user accounts are
+   automatically created the first time a user signs in with their SAML
+   credentials on the Mattermost server.
+
+Note that the standard SAML specification doesn't support a "lazy sync"
+option like AD/LDAP - instead, the SAML system is called every time the
+user authenticates in to Mattermost. Therefore, a user who has been
+deactivated in their identity provider will have access to Mattermost
+until their session expires and they need to log in again, or until
+`they are also deactivated in
+Mattermost <https://docs.mattermost.com/deployment/on-boarding.html#common-tasks>`__.
+
+Mattermost officially supports Okta and Microsoft ADFS as the identity providers (IDPs), 
+but you can also configure SAML for a custom IDP.
+
+- To set up SAML with Okta, see the `Okta guide <https://docs.mattermost.com/deployment/sso-saml-okta.html>`_
+- To set up SAML with Microsoft ADFS, see the `ADFS guide <https://docs.mattermost.com/deployment/sso-saml-adfs.html>`_
+
+Pre-installation
+----------------
+
+Before configuring SAML with Okta or Microsoft ADFS, make sure you have
+the `XML Security
+Library <https://www.aleksey.com/xmlsec/download.html>`__ installed on
+your Mattermost instance. The XML Security Library is usually included
+as part of Debian GNU/Linux.
+
+Also confirm if the ``xmlsec1-openssl`` library was successfully
+installed. If not, run 
+
+- ``apt-get install libxmlsec1-openssl`` on Ubuntu 
+- ``yum install xmlsec1-openssl`` on RHEL
+
+Troubleshooting
+---------------
+
+The following are troubleshooting suggestions on common error messages
+and issues.
+
+1. System Administrator locks themselves out of the system
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If the System Administrator is locked out of the system during SAML
+configuration process, they can set an existing account to System
+Administrator using `a commandline
+tool <http://docs.mattermost.com/deployment/on-boarding.html#creating-system-administrator-account-from-commandline>`__.
+
+2. Received error message: ``An account with that username already exists. Please contact your Administrator.``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This usually means an existing account has another authentication method enabled. If the user wants to use the existing account for SAML authentication, they should sign in using that method (such as email and password), then change their sign-in method to SAML via **Account Settings > Security > Sign-in method**.
+     
+This error message can also be received if the ``Username Attribute`` of their SAML credentials doesn't match the username of their Mattermost account. If so, the user can update the attribute at their identity provider (for instance, back to the old value if it had been previously updated). 
```

Comment:
It is -- the username is updated only if not already taken (confirmed with EN).

Reworded the section that an admin can update it (I wrote System Admin as they probably could update it would know the person who can)